File: ios/Kurir/modules/login/LoginViewModel.swift
//
//  AuthViewModel.swift
//  Kurir
//
//  Created by Loren on 19/8/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import MobileSDK
import Mobilex

class LoginViewModel: ViewModel {
    private let repository: AuthRepository?
    
    init(_ repository: AuthRepository?) {
        self.repository = repository
    }
    
    func login(name: String, password: String)-> LiveData<Auth?> {
        return LiveData<Auth?>() { [weak self] liveData in
            if let repository = self?.repository {
                repository.login(name: name, password: password, callback: { auth in
                    liveData.value = auth
                })
            }
        }
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/login/LoginViewModel.kt
package au.com.crazybean.mobilex.kurir.modules.login

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import au.com.crazybean.mobilex.data.model.Auth
import au.com.crazybean.mobilex.repository.auth.AuthRepository
import au.com.crazybean.sdk.mvvm.ViewModel

class LoginViewModel(private val repository: AuthRepository?) : ViewModel() {
    override fun onRelease() {
        super.onRelease()
        repository?.onRelease()
    }

    fun login(name: String, password: String): LiveData<Auth?> {
        return MutableLiveData<Auth?>().also { liveData ->
            repository?.login(name, password) {
                liveData.value = it
            }
        }
    }
}